code/modules/mob/living/carbon/xenomorph/castes/hivelord/abilities_hivelord.dm

```diff
@@ -410,7 +410,7 @@ GLOBAL_LIST_INIT(thickenable_resin, typecacheof(list(
 	#endif
 
 ///Called when the target xeno regains Sunder via heal_wounds in life.dm
-/datum/action/xeno_action/activable/healing_infusion/proc/healing_infusion_sunder_regeneration(datum/source, mob/living/carbon/xenomorph/patient)
+/datum/action/xeno_action/activable/healing_infusion/proc/healing_infusion_sunder_regeneration(datum/source, mob/living/carbon/xenomorph/patient, on_weeds = FALSE)
```

Comment:
dont add defaults like this for signal handlers it makes no sense.

also you can just check if you are on weeds because the signal source is the xeno you want to know if its on weeds.

Reply:
replace datum/source with mob/living/carbon/xenomorph/patient